### PR TITLE
MB-8152: Adds secondary pickup address to front end

### PR DESCRIPTION
**Pull request by kylehilltruss**
## Description

This pull request modifies the `MtoShipmentForm` component (and its associated data transformation/validation logic) to allow users to include a secondary pickup location for an HHG or NTS move, which either displays or hides a separate series of address based on a pair of radio buttons.

## Reviewer Notes

Make certain that creating a new shipment/updating an existing shipment's pickup address actually updates the associated fields in the database, and that viewing a shipment with an existing second pickup address defaults to the Yes/No radio button for having a second pickup address being set to "Yes," with the secondary address fields visible and populated.

## Setup

This component exists in storybook (`Customer Components`/`Forms`/`MtoShipmentForm`). It can also be seen by adding or editing a shipment from the third stage of the "Home" customer flow.

```sh
make server_run
make client_run
```

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

Closes [MB-8152](https://dp3.atlassian.net/browse/MB-8152).

## Screenshots

Before:

* https://user-images.githubusercontent.com/83614364/121259292-ee55ce00-c87d-11eb-85fc-1222952e9822.png

After:

* https://user-images.githubusercontent.com/83614364/121259319-f4e44580-c87d-11eb-9b35-d9f9ccc796ed.png
* https://user-images.githubusercontent.com/83614364/121259338-fa419000-c87d-11eb-9d33-4c935955d339.png

## Deployments

• https://my-milmove-pr-6770.mymove.sandbox.truss.coffee
• https://admin-milmove-pr-6770.mymove.sandbox.truss.coffee
• https://office-milmove-pr-6770.mymove.sandbox.truss.coffee
• https://prime-milmove-pr-6770.mymove.sandbox.truss.coffee

### CODE REVIEW

File: src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx [hunk 1]
```diff
@@ -53,13 +71,19 @@ class MtoShipmentForm extends Component {
...
+    const preformattedMtoShipment = {
       shipmentType: shipmentType || selectedMoveType,
       moveId,
       customerRemarks,
       pickup,
-      delivery: deliveryDetails,
-    });
+      delivery: {
+        ...delivery,
+        address: hasDeliveryAddress === 'yes' ? delivery.address : undefined,
```
Comment (YanZ777) [inline src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx:81]: It looks like you can take out lines 69-71 if you're using this logic here.

File: src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx [hunk 2]
```diff
@@ -44,7 +54,15 @@ class MtoShipmentForm extends Component {
...
 
-  submitMTOShipment = ({ shipmentType, pickup, hasDeliveryAddress, delivery, customerRemarks }) => {
+  submitMTOShipment = ({
+    shipmentType,
+    pickup,
+    hasDeliveryAddress,
+    delivery,
+    customerRemarks,
+    hasSecondaryPickup,
+    secondaryPickup,
```
Comment (YanZ777) [inline src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx:64]: Accessing this form from the Review page won't show the secondary pickup addresses. I ran into this issue as well when I was working on my story. I think you'll have to make a similar change to the `move.go` file. [See example here](https://github.com/transcom/mymove/pull/6772/files#diff-0cc5e5f9aaae7622547e5e79cffd7ba547d93d38dad5ba8a0a6ea0aab36ff1ac).

File: src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx (hunk 1 above)
Comment (kylehilltruss) [reply src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx:76]: Good point - fixed.

File: src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx (hunk 2 above)
Comment (kylehilltruss) [reply src/components/Customer/MtoShipmentForm/MtoShipmentForm.jsx:64]: Oh snap, thanks for catching that. Fixed.

**Review by cosburn — APPROVED** [review]
Looks great!

**Review by felipe-lee — APPROVED** [review]
LGTM

**Pull request merged**